src/libraries/System.IO.FileSystem/src/System/IO/Enumeration/FileSystemEntry.Unix.cs

```diff
@@ -36,8 +36,10 @@ namespace System.IO.Enumeration
 
             // IMPORTANT: Attribute logic must match the logic in FileStatus
 
+            entry._status = default;
+            entry._status.Invalidate();
```

Comment:
would it be possible to encapsulate the logic that is modifying the `FileStatus` below and move it into `FileStatus` itself? So the `FileStatus` could keep all the caching logic as it's private detail and return only a valid `FileStatus`?

```suggestion
            entry._status = FileStatus.Create(entry, directoryEntry, out FileAttributes attributes);
```

Reply:
Done.